framework/packages/abstract-client/src/client.rs

```diff
@@ -9,59 +41,93 @@ use crate::{
     publisher::{Publisher, PublisherBuilder},
 };
 
+/// Client to interact with Abstract accounts and modules
 pub struct AbstractClient<Chain: CwEnv> {
     pub(crate) abstr: Abstract<Chain>,
 }
 
+/// The result type for the Abstract Client.
 pub type AbstractClientResult<T> = Result<T, AbstractClientError>;
 
 impl<Chain: CwEnv> AbstractClient<Chain> {
+    /// Get abstract client from a chosen network. Abstract should be
+    /// already deployed on this chain
+    ///
+    /// ```
+    /// # use abstract_client::error::AbstractClientError;
+    /// # let sender = cosmwasm_std::Addr::unchecked("sender");
+    /// # let chain = cw_orch::prelude::Mock::new(&sender);
+    /// use abstract_client::client::AbstractClient;
+    ///
+    /// let client = AbstractClient::new(chain)?;
+    /// # Ok::<(), AbstractClientError>(())
+    /// ```
     pub fn new(chain: Chain) -> AbstractClientResult<Self> {
+        // TODO: New error type for not found abstract on this chain?
         let abstr = Abstract::load_from(chain)?;
```

Comment:
Loading abstract is completely unchecked, would we prefer to add check on `load_from` or here?